ext/GaussTuranQuadratureTaylorDiffExt.jl

```diff
@@ -0,0 +1,15 @@
+module GaussTuranQuadratureTaylorDiffExt
+using GaussTuranQuadrature: n_derivs, TaylorDiffIntegrand, evalrule
+import GaussTuranQuadrature: GaussTuranRule
+using TaylorDiff: derivatives, value
+
+function (I::GaussTuranRule)(f::TaylorDiffIntegrand)
+    (; a, b, W, X) = I
+    _f = f.f
+    order = f.order === nothing ? Val(n_derivs(I)) :
+            f.order isa Integer ? Val(f.order) :
+            f.order
+    __f = order isa Val{1} ? x -> (_f(x),) : x -> value(derivatives(_f, x, 1, order))
```

Comment:
Thanks. Some questions:
- Is this nested/chained use of ternary operators common practice? Then it is just something for me to get used to. Otherwise it would be better to write it out with `if else`.
- Would it be an idea to store `f.order` as a `Val`?